### DISCUSSION TITLE
Schedulers

### DISCUSSION THREAD

**Discussion by kiranmclernon** [mesa/mesa#1191] [General]
Hi all, I am new to Mesa but not to python
I was wondering if there were any methods to facilitate the combinations of different schedulers for different agents. This way a more complex scheduler could be created.
Apologies for the basic question but any help would be greatly appreciated.
Thanks

**Comment by trevorslab** [comment]
You can explore how to step by agent type.


https://mesa.readthedocs.io/en/latest/apis/time.html#mesa.time.RandomActivationByType

I did this a few years ago and had to alter the scheduler for a custom step
order similar to what you describe.  I don’t have code handy right now.  If
I find it I’ll send it along.
